pkg/api/v1/types.go

```diff
@@ -1709,6 +1714,15 @@ type SecretType string
 
 const (
 	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+
+	// SecretTypeDockercfg contains a dockercfg file that follows the same format rules as ~/.dockercfg
+	//
+	// Required fields:
+	// - Secret.Data["dockercfg"] - a serialized ~/.dockercfg file
+	SecretTypeDockercfg SecretType = "kubernetes.io/dockercfg"
+
+	// DockerConfigKey is the key of the required data for SecretTypeDockercfg secrets
+	DockerConfigKey = ".dockercfg"
```

Comment:
> And why is DockerConfigKey != SecretTypeDockercfg?

Because I don't like leading dots?  I like having the secret type qualified by "kubernetes.io" like `SecretTypeServiceAccountToken`, but I don't feel very strongly past that.  If you prefer "kubernetes.io/.dockercfg", I'm willing to change it, but I think it's more prone to typos.